### PR TITLE
changes to have the smeared file saved

### PR DESCRIPTION
**Pull request by mkilpatr**
Changes needed to smear the QCD

### CODE REVIEW

**Comment by mkilpatr** [comment]
There are now separate modules for the response files and qcd smearing files. I have added the nanoaod framework to allow QCD Smearing, response calculation, and base calculation in the same framework. The response creates on file. The QCD Smearing creates two files which are then hadded together in the background.

File: python/postprocessing/framework/postprocessor.py [hunk 1]
```diff
@@ -138,16 +161,21 @@ def run(self) :
...
 	
 	print  totEntriesRead/(time.clock()-t0), "Hz"
 
-
-	if self.haddFileName :
-		os.system("./haddnano.py %s %s" %(self.haddFileName," ".join(outFileNames))) #FIXME: remove "./" once haddnano.py is distributed with cms releases
+	if self.haddFileName:
+		if self.typeofprocess == "smear": Name = os.path.join(self.outputDir, os.path.basename(fname).replace(".root",outpostfix+"_smear.root"))
+		os.system("./haddnano.py %s %s" %(Name," ".join(outFileNames))) #FIXME: remove "./" once haddnano.py is distributed with cms releases
+		os.system("rm %s" %(" ".join(outFileNames)))
```
Comment (zhenbinwu) [inline python/postprocessing/framework/postprocessor.py:178]: Will these work on condor?

**Comment by mkilpatr** [comment]
I still need to test that part. It works locally though.

**Comment by mkilpatr** [comment]
Changed location of hadd file

**Comment by zhenbinwu** [comment]
So this branch is needed for QCD smearing. The code seems OK, merged to QCD branch

**Pull request merged**